src/elements/Step/Step.js

```diff
@@ -9,43 +9,36 @@ import StepGroup from './StepGroup'
 import StepTitle from './StepTitle'
 
 /** A step shows the completion status of an activity in a series of activities. */
-export default class Step extends Component {
-
-  static Content = StepContent
-  static Description = StepDescription
-  static Group = StepGroup
-  static Title = StepTitle
+function Step(props) {
+  const { active, className, children, completed, description, disabled, icon, href, link, onClick, title } = props
+  const classes = cx(
+    'ui',
+    useKeyOnly(active, 'active'),
+    useKeyOnly(completed, 'completed'),
+    useKeyOnly(disabled, 'disabled'),
+    useKeyOnly(link, 'link'),
+    className,
+    'step',
+  )
+  const rest = getUnhandledProps(Step, props)
+
+  const content = [
+    icon && iconPropRenderer(icon),
+    <StepContent {...{ children, description, title, ...rest }} />,
+  ]
```

Comment:
Most `content` component parts are only required if there is an adjacent icon or image.  Checking the Step markup, it looks like content is not included when there is [no icon](http://semantic-ui.com/elements/step.html#step).  This includes when there is only a [title and description](http://semantic-ui.com/elements/step.html#description).  In some cases, adding the `content` without an adjacent icon/image will break layouts.

Our current API spec does not allow mixing children when using smart props for markup.  This is because smart props generate children, including conditional `content` and other hierarchy.  There is then no sure fire way for us to know where to insert the user's children to prevent issues. So, the API here would  support one of these two cases but not both:

```jsx
<Step
  icon='truck'
  title='Shipping'
  description='Choose your shipping options'
/>

<Step>
  <Icon name='truck' />
  <Step.Content>
    <Step.Title>Shipping</Step.Title>
    <Step.Description>Choose your shipping options</Step.Description>
  </Step.Content>
</Step>
```

Lastly, the array children will need unique `key` props.  This might end up looking something more like this:

```jsx
const content = icon ? [
  iconPropRenderer(icon),
  (title || description) && <StepContent key='content' {...{ description, title }} />,
] : [
  title && <StepTitle key={'title'} />,
  description && <StepDescription key={'description'} />,
]
```

***

As an aside, I'm on the fence with allowing children as the "primary content" of a component while also using smart props.  This has been considered (and used in the past, Item.js) but under our current spec is not supported.  I haven't decided yet if it is good to support this, or if it just adds confusion and complexity.

Here is what it could look like:

```jsx
// title provided, use children as "primary content" (i.e. description)
<Step icon='truck' title='Shipping'>
  Choose your shipping options
</Step>
```

This looks nice and saves on markup, but would require a lot more conditional logic and propTypes to be done correctly.  For now, I think we should wait on it and just go with the very clear and explicit "use only smart props or use only children" approach.